io_scene_gltf/material.py

```diff
@@ -0,0 +1,263 @@
+    separator.location = -749, -130
+    links.new(inputNode.outputs[4], separator.inputs[0])
+
+    multRoughnessNode = tree.nodes.new('ShaderNodeMath')
+    multRoughnessNode.location = -476, -50
+    multRoughnessNode.operation = 'MULTIPLY'
+    links.new(separator.outputs[1], multRoughnessNode.inputs[0])
+    links.new(inputNode.outputs[3], multRoughnessNode.inputs[1])
+    roughnessOutputLink = multRoughnessNode.outputs[0]
+
+    multMetalnessNode = tree.nodes.new('ShaderNodeMath')
+    multMetalnessNode.location = -476, -227
+    multMetalnessNode.operation = 'MULTIPLY'
+    links.new(separator.outputs[2], multMetalnessNode.inputs[0])
+    links.new(inputNode.outputs[2], multMetalnessNode.inputs[1])
+    metalnessOutputLink = multMetalnessNode.outputs[0]
+
+    # First mix
+    mixNode1 = tree.nodes.new('ShaderNodeMixShader')
+    mixNode1.location = 226, 429
+
+    fresnelNode = tree.nodes.new('ShaderNodeFresnel')
+    fresnelNode.location = 14, 553
+    links.new(inputNode.outputs[6], fresnelNode.inputs[1])
+
+    diffuseNode = tree.nodes.new('ShaderNodeBsdfDiffuse')
+    diffuseNode.location = 14, 427
+    links.new(colorOutputLink, diffuseNode.inputs[0])
+    links.new(roughnessOutputLink, diffuseNode.inputs[1])
+    links.new(inputNode.outputs[6], diffuseNode.inputs[2])
+
+    glossyNode = tree.nodes.new('ShaderNodeBsdfGlossy')
+    glossyNode.location = 14, 289
+    links.new(roughnessOutputLink, glossyNode.inputs[1])
+    links.new(inputNode.outputs[6], glossyNode.inputs[2])
+
+    links.new(fresnelNode.outputs[0], mixNode1.inputs[0])
+    links.new(diffuseNode.outputs[0], mixNode1.inputs[1])
+    links.new(glossyNode.outputs[0], mixNode1.inputs[2])
+
+    # Second mix
+    mixNode2 = tree.nodes.new('ShaderNodeMixShader')
+    mixNode2.location = 406, 239
+
+    glossyNode2 = tree.nodes.new('ShaderNodeBsdfGlossy')
+    glossyNode2.location = 66, -114
+    links.new(colorOutputLink, glossyNode2.inputs[0])
+    links.new(roughnessOutputLink, glossyNode2.inputs[1])
+    links.new(inputNode.outputs[6], glossyNode2.inputs[2])
+
+    links.new(metalnessOutputLink, mixNode2.inputs[0])
+    links.new(mixNode1.outputs[0], mixNode2.inputs[1])
+    links.new(glossyNode2.outputs[0], mixNode2.inputs[2])
+
+    links.new(mixNode2.outputs[0], outputNode.inputs[0])
+
+    return tree
+
+
+def get_pbr_group(op):
+    if not op.pbr_group:
+        op.pbr_group = create_pbr_group()
+    return op.pbr_group
+
+
+def create_material(op, idx):
+    material = op.gltf['materials'][idx]
+    material_name = material.get('name', 'materials[%d]' % idx)
+    return create_material_from_properties(op, material, material_name)
+
+
+def create_material_from_properties(op, material, material_name):
+    pbr_metallic_roughness = material.get('pbrMetallicRoughness', {})
+
+    mat = bpy.data.materials.new(material_name)
+    op.materials[material_name] = mat
+    mat.use_nodes = True
+    tree = mat.node_tree
+    links = tree.links
+
+    for n in tree.nodes:
```

Comment:
Why is this necessary?

Reply:
It just gives us a clean slate to work on. A new material comes with some nodes already in it (a material and output node).